samcli/cli/cli_config_file.py

```diff
@@ -31,42 +31,60 @@ class TomlProvider:
     def __init__(self, section=None):
         self.section = section
 
-    def __call__(self, config_dir, config_env, cmd_names):
+    def __call__(self, config_path, config_env, cmd_names):
         """
         Get resolved config based on the `file_path` for the configuration file,
         `config_env` targeted inside the config file and corresponding `cmd_name`
         as denoted by `click`.
 
+        :param config_path: The path of configuration file.
         :param config_env: The name of the sectional config_env within configuration file.
         :param cmd_names list(str): sam command name as defined by click
         :returns dictionary containing the configuration parameters under specified config_env
         """
 
         resolved_config = {}
 
-        samconfig = SamConfig(config_dir)
+        # Use default sam config file name if config_path only contain the directory
+        samconfig = (
+            SamConfig(os.path.split(config_path)[0], os.path.split(config_path)[1])
```

Comment:
you can probably use pathlib for this.
pathlib.parent should give you the directory without needing to do splits.

Reply:
Contributer of SamConfig() is using strings of directory and filename. Here I think to pass in strings would be consistent.

Reply:
I reconsidered this and change to use pathlib to avoid some edge cases.